### DISCUSSION TITLE
rst:0x8 (TG1WDT_SYS_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)

### DISCUSSION THREAD

**Discussion by julianhyatt** [tbnobody/OpenDTU#1070] [Q&A]
Hi,
today my OpenDTU NRF kit from blinkyparts has arrived. I have soldered it like described in the manual. Powering it up shows only the openDTU! sign on the screen. Reading the serial monitor output tells me that the device crash and start rebooting

> 20:10:52.090 > Starting OpenDTU
> 20:10:52.090 > Initialize FS... done
> 20:10:52.093 > Reading configuration... done
> 20:10:52.210 > Reading PinMapping... found valid mapping done
> 20:10:52.236 > Initialize Network... done
> 20:10:52.349 > Setting Hostname... Initialize NTP... done
> 20:10:52.352 > Initialize SunPosition... done
> 20:10:52.354 > Initialize MqTT... done
> 20:10:52.357 > Initialize WebApi... done
> 20:10:52.366 > Initialize Display... done
> 20:10:52.692 > Initialize LEDs... done
> 20:10:52.693 > Check for default DTU serial... done
> 20:10:52.697 > Initialize Hoymiles interface... ets Jul 29 2019 12:21:46
> 20:10:53.316 > 
> 20:10:53.316 > rst:0x8 (TG1WDT_SYS_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
> 20:10:53.321 > configsip: 0, SPIWP:0xee
> 20:10:53.323 > clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
> 20:10:53.330 > mode:DIO, clock div:2
> 20:10:53.332 > load:0x3fff0030,len:1184
> 20:10:53.334 > load:0x40078000,len:13192
> 20:10:53.336 > load:0x40080400,len:3028
> 20:10:53.338 > entry 0x400805e4
> 20:10:53.939 > E (759) esp_core_dump_flash: No core du枀�partition found!
> 20:10:53.944 > E (759) esp_core_dump_flash: No core dump partition found!
> 20:10:53.963 > 
> 20:10:53.964 > Starting OpenDTU
> 20:10:53.966 > Initialize FS... done
> 20:10:53.967 > Reading configuration... done
> 20:10:54.085 > Reading PinMapping... found valid mapping done
> 20:10:54.110 > Initialize Network... done
> 20:10:54.223 > Setting Hostname... Initialize NTP... done
> 20:10:54.226 > Initialize SunPosition... done
> 20:10:54.229 > Initialize MqTT... done
> 20:10:54.231 > Initialize WebApi... done
> 20:10:54.240 > Initialize Display... done
> 20:10:54.566 > Initialize LEDs... done
> 20:10:54.568 > Check for default DTU serial... done
> 20:10:54.571 > Initialize Hoymiles interface... ets Jul 29 2019 12:21:46
> ....

So I have tried to compile it myself and let it flash to the device. I tryed the source from master and also checked out the latest tag. The compile and flash ends with success but the problems still the same. I have also tryed to flash the factory binary with same error.

Is there something I am doing wrong or could it be possible that the ESP32 is broken? Flashing and testing I do with board only. Or is this 
> E (759) esp_core_dump_flash: No core dump partition found!
the problem?

Thanks in advance!
Best regards

Julian

**Comment by tbnobody** [comment]
The core dump partition is not the problem. You could try the following:
* Remove the ESP from the PCB and check if it's still rebooting during the initialize of the RF interface. 
  * If you are using the latest firmware it should boot up without any issues even without any connected RF module
* Provide some detailed photos of the ESP and the PCB

**Reply by julianhyatt** [reply]
Hi, 
ty for the feedback. After it doesnt work on the PCB I have tried and flashed it without the PCB and it still wont work. 

![IMG20230624070925](https://github.com/tbnobody/OpenDTU/assets/54834203/c297b531-8769-4c87-a477-84e50a607122)
![IMG20230624070939](https://github.com/tbnobody/OpenDTU/assets/54834203/071ec81d-9cc7-4582-9c02-8ef3ff4ce48f)
![IMG20230624070953](https://github.com/tbnobody/OpenDTU/assets/54834203/f16b02bb-a1ae-4f75-9fb7-6c2b648f8577)
![IMG20230624071000](https://github.com/tbnobody/OpenDTU/assets/54834203/08320556-af99-461a-bb26-bad26dfbd0e6)

Some of the ESP pins was completely bent and the metallic case has one bump. Maybe the device is broken. I have already ordered 3 new ESP for testing purpose.

**Comment by julianhyatt** [comment]
Ok, I have tryed one of the new ESP32 and it starts normally, but the display keeps dark. Maybe the AzDelivery ESP32 got a different pin out I have to check

**Comment by tbnobody** [comment]
Did you upload the blinkyparts device profile json and have you selected the profile which enables the display?
(and please write a mail to [shop@blinkyparts.com](mailto:shop@blinkyparts.com) regarding replacement of the defect part(s))

**Comment by julianhyatt** [comment]
> Did you upload the blinkyparts device profile json and have you selected the profile which enables the display?

No, did not notice that I have to do it, Ty, I will try.

> and please write a mail to [shop@blinkyparts.com](mailto:shop@blinkyparts.com) regarding replacement of the defect part(s)

Not needed, the ESPs arent this expensive and I already have an replacement. They can use the money to support their projects :)

**Comment by julianhyatt** [comment]
Ok, I uploaded the blinkyparts_esp32.json profile here, but its not showing up.
<img width="1440" alt="Bildschirmfoto 2023-06-24 um 12 02 29" src="https://github.com/tbnobody/OpenDTU/assets/54834203/5dd56cce-113c-4ed4-8759-5540c0506bcc">

But I can activate some profiles here and make some changes
<img width="1364" alt="Bildschirmfoto 2023-06-24 um 12 01 19" src="https://github.com/tbnobody/OpenDTU/assets/54834203/e5c88f30-741c-4a28-9732-20d6c7fda2d5">

<img width="1355" alt="Bildschirmfoto 2023-06-24 um 12 01 43" src="https://github.com/tbnobody/OpenDTU/assets/54834203/6704cf4e-0d51-430f-be91-60ec83de781d">

But the display keeps black

**Comment by tbnobody** [comment]
You should see something like this in the hardware view:
![image](https://github.com/tbnobody/OpenDTU/assets/1753222/c4337a3b-df6e-4303-bbed-779f6adcafce)


If this is not the case, please doublecheck that you really uploaded the json file and not some github page sourcecode (which happens easily if you just make a right click on the file and choose "save as".

**Reply by julianhyatt** [reply]
Now I see them. It seemed I hat to refresh 
If I select NRF, LEDs, Display I will get the reboot problem. So possible the delivered ESP isnt broken

**Comment by tbnobody** [comment]
Then try to unplug the RF module or the display to determine when the reboot problem vanishes.

**Reply by julianhyatt** [reply]
Without NRF Module, same Problem. Since it comes via config i have to erase memory and reflash. The delivered ESP32 boots normal till I set the config.

**Comment by tbnobody** [comment]
Then just begin simple. 
* Clean flashed module
* Choose "NRF" profile
* Choose "NRF, Display" profile
* Choose "NRF, Display, LEDs" profile

A short on the led's could also lead to such problems for example. If you select the "NRF" profile only the NRF module gets activated and leds and display are still inactive. So you could doublecheck when the error occours.

**Reply by julianhyatt** [reply]
Ok, ty for the support.

I have tried it, but without success. I also have wired only the NRF like this: https://github.com/tbnobody/OpenDTU/blob/master/docs/Wiring_ESP32_Schematic.png but after changing config to only NRF it suddenly crashes.
Also I have tried ESP only with NRF Settings, same, it directly crashes.

I build it with "generic_esp32", with only "generic" it reboots directly. With "generic_esp32" the esp boots and I can connect to wifi. Viva wifi I have tried some config stuff like adding the hm-1500 serial, add my wifi credentials, ...
But when I change Hardware Config to NRF, it directly crashs
![192 168 178 10_settings_device](https://github.com/tbnobody/OpenDTU/assets/54834203/dded33e9-19c1-4ad4-a882-e02e73e9390e)

Is there something I am missing? For example should the HM-1500 be running, so that the dtu can initiate the connection? I try everything only with the DTU stuff.

**Comment by julianhyatt** [comment]
Ok, now I have a working setup. I have use one of the new esp32 I ordered yesterday and the NRF module I have used for my flipper nrf board. 
![IMG20230624185536](https://github.com/tbnobody/OpenDTU/assets/54834203/04deec16-3949-4c2a-992c-c5d58c0d0acf)
<img width="1329" alt="Bildschirmfoto 2023-06-24 um 18 55 51" src="https://github.com/tbnobody/OpenDTU/assets/54834203/b9c380e4-f377-4414-9f3e-956d4c62c08e">

If I replace the NRF, it still works. If I replace the ESP32 and flash same FW as I flashed to the other ESP, it wont work and keep rebooting. 

I think I will try to fix the old ESP this evening with the breadboard and if it wont work tomorrow, I think the ESP is broken.

Thanks for your help!
